addons/stock/tests/test_product.py

```diff
@@ -291,7 +298,8 @@ def test_product_qty_field_and_context(self):
             'location_id': main_loc.id,
         } for i in range(3)])
 
-        self.env['stock.quant'].search([('product_id', '=', self.product_3.id)]).unlink()
+        # TODO: should stock user be allowed to remove quants?
```

Comment:
Nope. Nobody should be able to delete quant. They are cleaned automatically by `_unlink_zero_quants`